src/serialmonitor/outputBuffer.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { performance } from "perf_hooks";
+import * as vscode from "vscode";
+
+export class BufferedOutputChannel implements vscode.Disposable {
+    private _buffer: string[];
+    private timer: NodeJS.Timer;
+    private _lastFlushTime: number;
+
+    public constructor(private outputCallback: (value: string) => void, private flushIntervalMs: number) {
+        this._buffer = [];
+        this.timer = setInterval(() => this.tryFlush(), this.flushIntervalMs);
+        this._lastFlushTime = Number.NEGATIVE_INFINITY;
+    }
+
+    public append(value: string) {
+        this.add(value);
+    }
+
+    public appendLine(value: string) {
+        this.add(value + "\n");
+    }
+
+    public dispose() {
+        clearInterval(this.timer);
```

Comment:
Flush the buffer during disposal. Otherwise someone could write to the buffer, then immediately dispose it and we'd drop the content.